source/views/settings/screens/api-test/api-test.tsx

```diff
@@ -0,0 +1,160 @@
+
+import {DebugView} from '../debug'
+import {syntaxHighlight} from './util/highlight'
+import {CSS_CODE_STYLES} from './util/highlight-styles'
+import {ChangeTextEvent} from '../../../../navigation/types'
+
+const styles = StyleSheet.create({
+	container: {
+		backgroundColor: c.white,
+		flex: 1,
+	},
+	error: {
+		padding: 10,
+		color: c.brickRed,
+	},
+	output: {
+		marginVertical: 3,
+		paddingRight: 4,
+		...Platform.select({
+			ios: iOSUIKit.bodyObject,
+			android: material.body1Object,
+		}),
+	},
+	emptySearch: {
+		flex: 1,
+		alignItems: 'center',
+		justifyContent: 'center',
+	},
+	emptySearchText: {
+		fontSize: 18,
+		color: c.semitransparentGray,
+		textAlign: 'center',
+		paddingTop: 20,
+		paddingBottom: 10,
+	},
+})
+
+type DisplayMode = 'raw' | 'parsed'
+
+export const APITestView = (): JSX.Element => {
+	let [searchPath, setSearchPath] = React.useState<string>('')
+	let path = useDebounce(searchPath.trim().toLowerCase(), 500)
+
+	let [displayMode, setDisplayMode] = React.useState<DisplayMode>('raw')
+
+	let {data, error} = useQuery({
+		queryKey: ['api-test', path],
+		queryFn: ({signal, queryKey: [_group, path]}) => {
+			if (!path) {
+				return ''
+			}
+			return client.get(path, {signal, cache: 'no-store'}).text()
+		},
+		staleTime: 0,
+		cacheTime: 0,
+	})
+
+	let navigation = useNavigation()
+
+	React.useLayoutEffect(() => {
+		navigation.setOptions({
+			headerSearchBarOptions: {
+				autoCapitalize: 'none',
+				barTintColor: c.white,
+				// android-only
+				autoFocus: true,
+				hideNavigationBar: false,
+				hideWhenScrolling: false,
+				onChangeText: (event: ChangeTextEvent) =>
+					setSearchPath(event.nativeEvent.text),
+				placeholder: 'path/to/uri',
+			},
+		})
+	}, [navigation])
+
+	const JSONView = (): JSX.Element => {
+		if (data === undefined) {
+			return <></>
+		}
+
+		const parsed = JSON.parse(data ?? '')
+		const formatted = JSON.stringify(parsed, null, 2)
+		const highlighted = syntaxHighlight(formatted)
+
+		const HTML_CONTENT = `
+			${CSS_CODE_STYLES}
+			<pre>${highlighted}</pre>
+		`
+
+		return <HtmlContent html={HTML_CONTENT} />
+	}
+
+	const EmptySearch = () => {
+		return (
+			<View style={styles.emptySearch}>
+				<Icon
+					color={c.semitransparentGray}
+					name="analytics-outline"
+					size={64}
+				/>
+				<Text style={styles.emptySearchText}>Search for an API Endpoint</Text>
+			</View>
+		)
+	}
+
+	let APIResponse =
+		error !== null ? (
+			<TextInput
+				editable={false}
+				// this aligns the text to the top on iOS, and centers it on Android
+				multiline={true}
+				scrollEnabled={true}
+				style={[styles.output, styles.error]}
+				// use multiline with textAlignVertical="top" for the same behavior in both platforms
+				textAlignVertical="top"
+				value={String(error)}
+			/>
+		) : displayMode === 'raw' ? (
+			path ? (
+				<JSONView />
+			) : (
+				<EmptySearch />
+			)
+		) : (
+			<DebugView state={JSON.parse(data || '{}')} />
+		)
```

Comment:
```suggestion
		) : !path ? (
			<EmptySearch />
		) : displayMode === 'raw' ? (
			<JSONView />
		) : (
			<DebugView state={JSON.parse(data || '{}')} />
		)
```

is this valid?

Reply:
I like that a lot, it keeps the json/list empty views in-sync. Thanks!